pages/repo/preview.js

```diff
@@ -0,0 +1,81 @@
+import React, { useRef } from 'react'
+import { withRouter } from 'next/router'
+import { graphql, compose } from 'react-apollo'
+import gql from 'graphql-tag'
+import { ColorContextProvider } from '@project-r/styleguide'
+import { renderMdast } from 'mdast-react-render'
+
+import Loader from '../../components/Loader'
+import Frame from '../../components/Frame'
+import { getSchema } from '../../components/Templates'
+
+import * as fragments from '../../lib/graphql/fragments'
+import initLocalStore from '../../lib/utils/localStorage'
+import withT from '../../lib/withT'
+
+const getCommitById = gql`
+  query getCommitById($repoId: ID!, $commitId: ID!) {
+    repo(id: $repoId) {
+      ...EditPageRepo
+      commit(id: $commitId) {
+        ...CommitWithDocument
+      }
+    }
+  }
+  ${fragments.EditPageRepo}
+  ${fragments.CommitWithDocument}
+`
+
+const PreviewPage = ({ dark, router, data = {} }) => {
+  const {
+    loading,
+    error,
+    repo: {
+      commit: { document }
+    }
+  } = data
+  const { repoId, commitId } = router.query
+  const storeKey = [repoId, commitId].join('/')
+  const store = initLocalStore(storeKey)
+  let localState = store.get('editorState')
```

Comment:
@annatraussnig  Do you think we need `this.checkLocalStorageSupport()` here as well: https://github.com/orbiting/publikator-frontend/blob/4f617ff728bc33188a59073225b815d51da21b5e/pages/repo/edit.js#L577

Reply:
I don't think it brings much. `initLocalStore` works either way, the data is just not persisted if if `localStorage` is not supported. That's acceptable. Plus writers get a warning about it in `edit.js` (I know that's not much of an argument, but one cannot say that it comes as a total surprise)